xbmc/peripherals/EventScanner.cpp

```diff
@@ -69,31 +69,31 @@ void CEventScanner::Release(CEventRateHandle* handle)
 {
   CSingleLock lock(m_mutex);
 
-  const float oldRate = GetRateHz();
+  const double oldRate = GetRateHz();
 
   m_handles.erase(std::remove_if(m_handles.begin(), m_handles.end(),
     [handle](const EventRateHandle& myHandle)
     {
       return handle == myHandle.get();
     }), m_handles.end());
 
-  const float newRate = GetRateHz();
+  const double newRate = GetRateHz();
 
   CLog::Log(LOGDEBUG, "PERIPHERALS: Event sampling rate set from %.2f to %.2f", oldRate, newRate);
 }
 
 void CEventScanner::Process(void)
 {
-  float nextScanMs = static_cast<float>(SystemClockMillis());
+  double nextScanMs = static_cast<double>(SystemClockMillis());
 
   while (!m_bStop)
   {
     m_scanEvent.Reset();
 
     m_callback->ProcessEvents();
 
-    const float nowMs = static_cast<float>(SystemClockMillis());
-    const float scanIntervalMs = GetScanIntervalMs();
+    const double nowMs = static_cast<double>(SystemClockMillis());
+    const double scanIntervalMs = GetScanIntervalMs();
```

Comment:
consider using xbmcThreads::EndTime. it handles wrap-arounds for you which is a notorious problem on windows (it loops after 40h or so).